Code/Framework/AzCore/AzCore/DOM/DomBackendRegistry.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/DOM/DomBackendRegistryInterface.h>
+#include <AzCore/std/containers/unordered_map.h>
+
+namespace AZ::DOM
+{
+    //! DOM backend registry implementation.
+    //! \see BackendRegistryInterface
+    class BackendRegistry final : public BackendRegistryInterface
+    {
+    public:
+        AZ_RTTI(BackendRegistry, "{95533861-6201-4E45-BD03-097A20850C48}", BackendRegistryInterface);
+        AZ_CLASS_ALLOCATOR(BackendRegistry, AZ::SystemAllocator, 0);
+
+        AZStd::unique_ptr<Backend> GetBackendByName(AZStd::string_view name) override;
+        AZStd::unique_ptr<Backend> GetBackendForExtension(AZStd::string_view extension) override;
```

Comment:
Since these functions don't get an existing instance but use a factory to create a new backend, consider renaming these to:
```
CreateBackendByName
CreateBackendForExtension
```